balance.py

```diff
@@ -497,27 +510,67 @@ def callback_teams(self, players, channel):
 
     def cmd_do(self, player, msg, channel):
         """Forces a suggested switch to be done."""
-        if self.suggested_pair:
-            self.execute_suggestion()
+        if not self.suggested_pair:
+            return
+        
+        if not self.switch_opt_in:
+            return
+
+        self.execute_suggestion()
+
+    def cmd_dont(self, player, msg, channel):
+        """Vetoes a suggested switch to be done."""
+        if not self.suggested_pair:
+            return
+
+        if self.switch_opt_in:
+            return
+
+        self.msg("An admin prevented the switch! The switch will be terminated.")
+        self.suggested_pair = None
+        self.suggested_agree = [False, False]
 
     def cmd_agree(self, player, msg, channel):
         """After the bot suggests a switch, players in question can use this to agree to the switch."""
-        if self.suggested_pair and not all(self.suggested_agree):
-            p1, p2 = self.suggested_pair
+        if not self.suggested_pair or not self.switch_opt_in or all(self.suggested_agree):
+            return
```

Comment:
I would move swtich_opt_in checking to top of cmd_agree function. To indicate that this command is not used on auto switch enabled.